### PR TITLE
show & hide mapnik layers

### PR DESCRIPTION
**Pull request by dgaubert**
Implements [Windshaft-cartodb#550](https://github.com/CartoDB/Windshaft-cartodb/issues/550). Allows to show/hide layers by requesting the different layer indexes in the URL.

*Note*: It's still pending to use layer id as an alternative to current layer index

### CODE REVIEW

**Comment by dgaubert** [comment]
Mapnik layerId filtering is done. 

I think we can CR.

**Comment by dgaubert** [comment]
I fixed layer interactivity issue. I'm ready for CR. 

cc @rochoa

File: lib/windshaft/models/mapconfig.js [hunk 1]
```diff
@@ -127,6 +127,26 @@ function getLayerId(rawMapConfig, layerIndex) {
     return layerId;
 }
 
+MapConfig.prototype.getLayerById = function (layerId) {
```
Comment (rochoa) [inline lib/windshaft/models/mapconfig.js]: I don't find any usage of this one.

File: lib/windshaft/models/mapconfig.js [hunk 2]
```diff
@@ -127,6 +127,26 @@ function getLayerId(rawMapConfig, layerIndex) {
...
+    var layers = this.getLayers();
+
+    for (var i = 0; i < layers.length; i++) {
+        if (layers[i].id === layerId) {
+            return layers[i];
+        }
+    }
+};
+
+MapConfig.prototype.getIndexByLayerId = function (layerId) {
```
Comment (rochoa) [inline lib/windshaft/models/mapconfig.js]: Should we return `-1` when the layer is not found? As when you use `Array.prototype.indexOf`.

**Pull request merged**